src/allocator.rs

```diff
@@ -39,14 +39,14 @@ impl NodePtr {
                     panic!("unknown NodePtr type");
                 }
             },
-            (self.0 & NODE_PTR_IDX_MASK) as usize,
+            (self.0 & NODE_PTR_IDX_MASK),
         )
     }
 
     pub(crate) fn as_index(&self) -> usize {
         match self.node_type() {
-            (ObjectType::Pair, idx) => idx * 2,
-            (ObjectType::Bytes, idx) => idx * 2 + 1,
+            (ObjectType::Pair, idx) => (idx as usize) * 2,
+            (ObjectType::Bytes, idx) => (idx as usize) * 2 + 1,
```

Comment:
Do we want `u64` here? Is it possible that `usize` can be a `u32` or is it guaranteed to be `u64` or bigger?

Reply:
I guess `[]` indexing uses `usize` so if `usize` is `u32` we are already in trouble.